Use a bigger safety buffer when estimating TX prices

ldmberman opened the pull request:
The buffer is the same to that used before the switch to the linear difficulty. A more optimistic estimation does not perform well under the present difficulty fluctuations.

llaisdy commented:
Looks good by eye.

Is `MaxDiff - (MaxDiff - CurrentDiff) * 2` equivalent to old-style `Diff - 2`?

Minor quibble: I might export `twice_smaller_diff` (maybe even from `ar_difficulty.erl`) and use it in `ar_http_iface_middleware.erl` as well as in the tests module.

ldmberman commented:
@llaisdy equivalent to old-style `Diff - 1`, yes. I added a fixup, also removed the confusing `Height` argument from `ar_mine:max_diifficulty`, which does not depend on it.
